feat(next): context.isPrerendered (#9301)
withastro/docs: src/content/docs/en/reference/api-reference.mdx

See also: [`context.routePattern`](#contextroutepattern)

### `Astro.isPrerendered`

<p>

> **Type**: `string`<br />

Reviewer: Are these strings or boolean?

Reviewer 2: Good catch, I read too quickly 😓 ([`boolean`](https://github.com/withastro/astro/blob/2fe97bc4fd855ca559664c578ace3f410134eae5/packages/astro/src/types/public/context.ts#L346)).

Author: copy pasting...my bad!

```suggestion
**Type**: `boolean`<br />
```

<Since v="5.0.0" />
</p>

A boolean representing whether or not the current page is prerendered.

> You can use this property to run conditional logic in middlewares, eg. to avoid accessing headers in prerendered pages.

Reviewer: ```suggestion
You can use this property to run conditional logic in middlewares, for example, to avoid accessing headers in prerendered pages.
```
I'm not sure about the language of "avoid accessing headers".... that makes it sound like accessing headers a bad thing and you want to avoid it at all costs. Is that true? Or, is it that you *can't* access headers in prerendered pages? Is this a helpful alternative for something you can't otherwise do?

Just trying to get a sense of how this is useful/helpful to know how best to phrase it!

Author: Not sure how to phrase this! An example: it's useless to check if a user is authenticated (eg. using a cookie which uses headers) if you generate a static og image. Also, I found out today that stuff can break if you use headers in this situation so maybe "avoid" is the right term

Reviewer: OK, then I'm happy to go with this!

See also: [`Astro.routePattern`](#astroroutepattern)

> ### `Astro.isPrerendered`

Reviewer 2: Except this:
```suggestion
### `context.isPrerendered`
```
Looks good to me!

<p>

> **Type**: `string`<br />

Author: ```suggestion
**Type**: `boolean`<br />
```
